### ISSUE TITLE
-ContinueOnError does not work

### ISSUE THREAD

**Issue by GreatTeacherBasshead** [psake/psake#295]
`script.ps1`:
```
Task Default -Depends T1

Task T1 -ContinueOnError {
    Write-Error "Task 1"
}
```

## Expected Behavior
```
Error in task T1...
psake succeeded executing...
```

## Current Behavior
`Invoke-psake script.ps1`:
```
Executing T1
----------------------------------------------------------------------
Error in task T1. task 1
----------------------------------------------------------------------
Error:
At C:\Program Files\WindowsPowerShell\Modules\psake\4.9.0\public\Assert.ps1:69 char:9 +
throw ('Assert: {0}' -f $failureMessage) +
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ [<<==>>] Exception: Assert: Call stack was corrupt. Expected t1, but got default.
```

`Invoke-psake psake.ps1 -taskList T1`
```
Executing T1
----------------------------------------------------------------------
Error in task T1. task 1
----------------------------------------------------------------------
Error:
At C:\Program Files\WindowsPowerShell\Modules\psake\4.9.0\public\Invoke-psake.ps1:303 char:25 +
invoke-task $task +
~~~~~~~~~~~~~~~~~ [<<==>>] Exception: Exception calling "Pop" with "0" argument(s): "Stack empty." --> Stack empty.
```

## Steps to Reproduce (for bugs)
Invoke script.ps1

## Context
Upgrading psake from version 4.7.4 to 4.9.0. Works fine in 4.7.4.

## Your Environment
* Module version used: 4.9.0
* Operating System: Win10/Win2016
* PowerShell: 5.1

**Comment by UberDoodles** [comment]
I think I've identified the source of this issue.

At the end of the Invoke-Task function (line 154), it pops what it expects to be the current task off the call stack.

However, if -ContinueOnError is specified, this triggers a code block earlier in the function which also pops the current task off the call stack (line 132). So it ends up popping two tasks instead of one.

It's a nice quick fix; just need to remove the extra pop in the -ContinueOnError code. I'll see if I can get a pull-request raised for this tonight. Depends how long it takes me to put write a unit test.

**Comment by UberDoodles** [comment]
I've fixed this on a branch in my fork now and added a Pester test. But I've hit a slight hiccough; it caused another test to fail. The test relates to the shared task functionality.

So it looks like that extra pop was added in this changeset, which is the commit which introduced the shared task functionality, and the same commit in which that Pester test was added : https://github.com/psake/psake/commit/d1b925520ac08f87b981cf2f8677473317d2a56b#diff-1cf57fc02f27014423b1f80dfae0d708

I need to have a closer look at the shared task functionality, which I'm not very familiar with, and try to figure out how to update the code such that it will accommodate both scenarios.
